color_refinement.py

```diff
@@ -78,6 +79,60 @@ def color_refine(coloring: "Coloring") -> "Coloring":
     return coloring
 
 
+def fast_color_refine(graph: "Graph"):
+    # takes initial coloring
+    qlist = DoubleLinkedList()
+    color_map = {}
+    neigh_map = {}
+    #TODO: Replace this with better things, because it's now just used to get the amount of colors
+    # because I did not figure out how to do this with the Coloring class yet
+    # Neigh_map might not be neccessary
+    for v in graph.vertices:
+        if v.colornum not in color_map.keys():
+            color_map[v.colornum] = []
+        color_map[v.colornum].append(v)
+        if v not in neigh_map.keys():
+            neigh_map[v] = None
+        neigh_map[v] = v.neighbours
```

Comment:
Dit kun je toch direct doen? Want die map is leeg en een vertex komt maar een keer langs, dus je weet dat ie er nog nooit in zit

Reply:
Ja precies :) Ik was vrijdag niet helemaal helder en ik wist dat dit sowieso zou werken, maar zoals Dorien hier boven al zei is de vraag of dit tijdswinst oplevert, dus om het te versimpelen sloop ik dit eruit